src/kademlia/dht_storage.cpp

```diff
@@ -532,6 +556,36 @@ namespace
 			if (!peers.empty() && peers.capacity() / peers.size() >= 4u)
 				peers.shrink_to_fit();
 		}
+
+		void refresh_infohashes_sample()
```

Comment:
I need `infohashes_sample.sample` to be a string due to performance reasons. This way that data is added as is to `entry["samples"]` without the need of conversion to the "blob" string each time. I could refactor a little though.